### ISSUE TITLE
trust/asn1 uses deprecated tasn1 macros

### ISSUE THREAD

**Issue by dmacks** [p11-glue/p11-kit#453]
Building p11-kit-0.24.1 on OS X 10.13 with libtasn1-4.18.0 (via fink), tasn1 is correctly detected:
```
% ./configure
...
checking for LIBTASN1... yes
checking for asn1Parser... /sw/bin/asn1Parser
...
   With libtasn1 dependency:        yes
...
```
but gives a mile of compiler warnings of the type:
```
  CC       trust/p11_kit_trust_la-builder.lo
In file included from trust/builder.c:40:
./trust/asn1.h:46:1: warning: 'node_asn' macro is deprecated, use 'asn1_node' instead. [-W#pragma-messages]
node_asn *       p11_asn1_decode                    (p11_dict *asn1_defs,
^
/sw/include/libtasn1.h:524:21: note: expanded from macro 'node_asn'
#   define node_asn _Pragma ("GCC warning \"'node_asn' macro is deprecated, use 'asn1_node' instead.\"") asn1_node_st
                    ^
<scratch space>:87:6: note: expanded from here
 GCC warning "'node_asn' macro is deprecated, use 'asn1_node' instead."
     ^
In file included from trust/builder.c:40:
./trust/asn1.h:52:1: warning: 'node_asn' macro is deprecated, use 'asn1_node' instead. [-W#pragma-messages]
node_asn *       p11_asn1_create                    (p11_dict *asn1_defs,
^
/sw/include/libtasn1.h:524:21: note: expanded from macro 'node_asn'
#   define node_asn _Pragma ("GCC warning \"'node_asn' macro is deprecated, use 'asn1_node' instead.\"") asn1_node_st
                    ^
<scratch space>:89:6: note: expanded from here
 GCC warning "'node_asn' macro is deprecated, use 'asn1_node' instead."
     ^
```

**Comment by ueno** [comment]
I think it has been fixed in #427, after 0.24.1 release. Could you perhaps check the latest git version to see if all the deprecated warnings are gone?

**Comment by dmacks** [comment]
As a downstream packager, I'm fine waiting a month for a 0.25 that has the same SONAME. From my perspective, deprecation warnings are noise but not breakage, so no actual harm.
